src/xwayland.rs

```diff
@@ -501,6 +501,8 @@ pub trait Primary {
     fn remove_baselayer_window(&self) -> Result<(), Box<dyn std::error::Error>>;
     /// Request a screenshot from Gamescope
     fn request_screenshot(&self) -> Result<(), Box<dyn std::error::Error>>;
+    /// Sets the display mode control for Gamescope
+    fn set_mode_control(&self, width: u32, height: u32, super_res: u32) -> Result<(), Box<dyn std::error::Error>>;
```

Comment:
This will also need to be changed along with the implementation below.

```suggestion
    /// Sets the display mode control for Gamescope
    fn set_mode_control(
        &self,
        xwayland_id: u32,
        width: u32,
        height: u32,
        super_res: u32,
    ) -> Result<(), Box<dyn std::error::Error>>;
```